wyzesense2mqtt.py

```diff
@@ -12,169 +10,266 @@
-
-    signal_strength_sensor_topic = "{0}sensor/wyzesense_{1}_signal_strength/config".format(HASS_TOPIC_ROOT, sensor_mac)
-    client.publish(signal_strength_sensor_topic, payload = None, qos = MQTT_QOS, retain = MQTT_RETAIN)
+def clear_topics(sensor_mac):
+    global CONFIG
+    LOGGER.info("Clearing sensor topics")
+    state_topic = "{0}{1}".format(CONFIG['wyzesense2mqtt_topic_root'], sensor_mac)
+    MQTT_CLIENT.publish(state_topic, payload = None, qos = CONFIG['mqtt_qos'], retain = CONFIG['mqtt_retain'])
+
+    entity_types = ['state', 'signal_strength', 'battery']
+    for entity_type in entity_types:
+        sensor_type = ("binary_sensor" if entity_type == "state" else "sensor")
+        entity_topic = "{0}{1}/wyzesense_{2}/{3}/config".format(CONFIG['hass_topic_root'], sensor_type, sensor_mac, entity_type)
+        MQTT_CLIENT.publish(entity_topic, payload = None, qos = CONFIG['mqtt_qos'], retain = CONFIG['mqtt_retain'])
+
+def on_connect(MQTT_CLIENT, userdata, flags, rc):
+    global CONFIG
+    LOGGER.info("Connected to mqtt with result code {0}".format(str(rc)))
+    MQTT_CLIENT.subscribe([(SCAN_TOPIC, CONFIG['mqtt_qos']), (REMOVE_TOPIC, CONFIG['mqtt_qos'])])
+    MQTT_CLIENT.message_callback_add(SCAN_TOPIC, on_message_scan)
+    MQTT_CLIENT.message_callback_add(REMOVE_TOPIC, on_message_remove)
+
+def on_disconnect(MQTT_CLIENT, userdata, rc):
+    LOGGER.info("Disconnected from mqtt with result code {0}".format(str(rc)))
+    MQTT_CLIENT.message_callback_remove(SCAN_TOPIC)
+    MQTT_CLIENT.message_callback_remove(REMOVE_TOPIC)
 
-    battery_sensor_topic = "{0}sensor/wyzesense_{1}_battery/config".format(HASS_TOPIC_ROOT, sensor_mac)
-    client.publish(battery_sensor_topic, payload = None, qos = MQTT_QOS, retain = MQTT_RETAIN)
-
-def on_event(ws, event):
-    _LOGGER.info("Processing Event")
-    _LOGGER.debug("Event data: {0}".format(event))
-    if event.Type == "state":
+# Process messages
+def on_message(MQTT_CLIENT, userdata, msg):
+    LOGGER.info("{0}: {1}".format(msg.topic, str(msg.payload)))
+
+# Process message to scan for new sensors
+def on_message_scan(MQTT_CLIENT, userdata, msg):
+    global SENSORS
+    LOGGER.info("In on_message_scan: {0}".format(msg.payload.decode()))
+
+    try:
+        result = WYZESENSE_DONGLE.Scan()
+        LOGGER.debug("Scan result: {0}".format(result))
+        if result:
+            sensor_mac, sensor_type, sensor_version = result
+            sensor_type = ("motion" if sensor_type == 2 else "opening")
+            if valid_sensor_mac(sensor_mac):
+                if SENSORS.get(sensor_mac) is None:
+                    add_sensor_to_config(sensor_mac, sensor_type, sensor_version)
+                    send_discovery_topics(sensor_mac)
+            else:
+                LOGGER.debug("Invalid sensor found: {0}".format(sensor_mac))
+        else:
+            LOGGER.debug("No new sensor found")
+    except:
+        pass
```

Comment:
Are you intentionally ignoring errors for some reason?

Reply:
See next commit, flake8 caught this and wanted explicit excepts. I am ignoring TimeoutError's as the WyzeSensePy library was generating them on occasion even though things were working. It was causing the MQTT client thread to bomb and resulting in only have the program working after those actions occurred.